src/main/java/internet/shop/dao/ShoppingCartDao.java

```diff
@@ -0,0 +1,20 @@
+package internet.shop.dao;
+
+import internet.shop.model.Product;
+import internet.shop.model.ShoppingCart;
+import java.util.List;
+import java.util.Optional;
+
+public interface ShoppingCartDao {
+    ShoppingCart create(ShoppingCart shoppingCart);
+
+    Optional<ShoppingCart> get(Long id);
+
+    List<ShoppingCart> getAll();
+
+    boolean update(ShoppingCart shoppingCart);
+
+    List<Product> getProducts(ShoppingCart shoppingCart);
```

Comment:
do we need this method on dao layer, can we just call shoppingCart.getProducts() ?